include/shared.h

```diff
@@ -61,8 +62,24 @@ bool hc_string_is_digit (const char *s);
 void hc_string_trim_trailing (char *s);
 void hc_string_trim_leading (char *s);
 
-size_t hc_fread (void *ptr, size_t size, size_t nmemb, FILE *stream);
-size_t hc_fwrite (const void *ptr, size_t size, size_t nmemb, FILE *stream);
+bool   hc_fopen    (HCFILE *fp, const char *path, char *mode);
+size_t hc_fread    (void *ptr, size_t size, size_t nmemb, FILE *stream);
+void   hc_fwrite   (const void *ptr, size_t size, size_t nmemb, FILE *stream);
+int    hc_fscanf   (HCFILE *fp, const char *format, void *ptr);
+int    hc_fprintf  (HCFILE *fp, const char *format, ...);
+int    hc_vfprintf (HCFILE *fp, const char *format, va_list ap);
+int    hc_fseek    (HCFILE *fp, off_t offset, int whence);
+void   hc_rewind   (HCFILE *fp);
+off_t  hc_ftell    (HCFILE *fp);
+int    hc_fgetc    (HCFILE *fp);
+int    hc_fputc    (int c, HCFILE *fp);
+char  *hc_fgets    (char *buf, int len, HCFILE *fp);
+int    hc_fileno   (HCFILE *fp);
+int    hc_feof     (HCFILE *fp);
+void   hc_fflush   (HCFILE *fp);
+void   hc_fclose   (HCFILE *fp);
+size_t hc_fwrite_compress (void *ptr, size_t size, size_t nmemb, HCFILE *fp);
```

Comment:
Why is there still _compress functions? Please remove. If this is related to potfile problem, then please explain the error in potfile and how to reproduce. Then update PR so that error occurs. I'll then try to fix.

Reply:
Because the porting it's not end. If you want check the potfile bug you need switch to new hc_* functions (now are all commented, check potfile source)

Reply:
You need also update the "struct potfile_ctx" to test it